fv3core/stencils/c2l_ord.py

```diff
@@ -74,61 +73,72 @@ def ord4_transform(
         va = a21 * utmp + a22 * vtmp
 
 
-def compute_cubed_to_latlon(
-    u: Quantity,
-    v: Quantity,
-    ua: FloatField,
-    va: FloatField,
-    comm: CubedSphereCommunicator,
-    do_halo_update: bool,
-):
+class CubedToLatLon:
     """
-    Interpolate D-grid to A-grid winds at latitude-longitude coordinates.
-
-    Args:
-        u: x-wind on D-grid (in)
-        v: y-wind on D-grid (in)
-        ua: x-wind on A-grid (out)
-        va: y-wind on A-grid (out)
-        comm: Cubed-sphere communicator
-        do_halo_update: If True, performs a halo update on u and v
+    Fortan name is c2l_ord2
     """
-    do_halo_update = do_halo_update and global_config.get_do_halo_exchange()
-    grid = spec.grid
 
-    if spec.namelist.c2l_ord == 2:
-        c2l_ord2(
-            u.storage,
-            v.storage,
-            grid.dx,
-            grid.dy,
-            grid.a11,
-            grid.a12,
-            grid.a21,
-            grid.a22,
-            ua,
-            va,
-            origin=grid.compute_origin(
-                add=(-1, -1, 0) if do_halo_update else (0, 0, 0)
-            ),
-            domain=grid.domain_shape_compute(
-                add=(2, 2, 0) if do_halo_update else (0, 0, 0)
-            ),
-        )
-    else:
-        if do_halo_update:
-            comm.vector_halo_update(u, v, n_points=grid.halo)
-        ord4_transform(
+    def __init__(self, grid, do_halo_update: bool):
+        """
+        Args:
+            grid: fv3core grid object
+            do_halo_update: If True, performs a halo update on u and v
+        """
+        self._do_halo_update = do_halo_update and global_config.get_do_halo_exchange()
```

Comment:
This looks like dangerous logic to put into the init. Should `do_halo_update` be an Optional[bool] and if it is None we grab the global_config?